Source/Data/Remotes/ParleyRemote.swift

```diff
@@ -4,9 +4,60 @@ import ObjectMapper
 import UIKit
 import Foundation
 
+
+public typealias ParleyRemoteInterceptor = ParleyRemoteRequestInterceptorProtocol & ParleyRemoteResponseInterceptorProtocol
+
+public protocol ParleyRemoteRequestInterceptorProtocol: AnyObject {
+    func httpRequestWillStart(with request: inout URLRequest) throws
+}
+
+public protocol ParleyRemoteResponseInterceptorProtocol: AnyObject {
+    func httpResponseReceived(response: HTTPURLResponse) throws
```

Comment:
What would be the need for intercepting the network response of the requests?

Reply:
We would need to check it the response hasn't been tampered.